inc/commondbtm.class.php

```diff
@@ -5456,6 +5456,51 @@ public function getTypeForeignKeyField(): ?string {
       return $type_class !== null ? $type_class::getForeignKeyField() : null;
    }
 
+   public function hasItemtypeOrModelPictures(array $picture_fields = ['picture_front', 'picture_rear', 'pictures']): bool {
+      $itemtype = $this->getType();
+
+      $has_pictures = false;
+
+      foreach ($picture_fields as $picture_field) {
+         if ($this->isField($picture_field)) {
+            if ($picture_field === 'pictures') {
+               $urls = importArrayFromDB($this->fields[$picture_field]);
+               if (!empty($urls)) {
+                  $has_pictures = true;
+                  break;
+               }
+            } else if (!empty($this->fields[$picture_field])) {
+               $has_pictures = true;
+               break;
+            }
+         } else {
+            $modeltype = $itemtype . "Model";
+            if (class_exists($modeltype)) {
+               /** @var CommonDBTM $model */
```

Comment:
A check should be done here to validate that `$this->fields[$fk]` is not empty. Indeed, it would prevent multiple requests to be done for assets that are not linked to a model.